code/game/objects/machinery/cryopod.dm

```diff
@@ -204,18 +204,13 @@
 /mob/living/proc/despawn(obj/machinery/cryopod/pod, dept_console = CRYO_REQ)
 
 	//Handle job slot/tater cleanup.
-	if(job in GLOB.jobs_regular_all)
-		var/datum/job/J = SSjob.name_occupations[job]
-		J.current_positions--
-		if((J.title in GLOB.jobs_regular_all) && isdistress(SSticker?.mode))
-			var/datum/game_mode/distress/D = SSticker.mode
-			var/datum/job/rank = SSjob.GetJob(job)
-			D.latejoin_larvapoints -= rank.larvaworth //Cryoing someone removes a player from the round, blocking further larva spawns until accounted for
-		if(J.title in GLOB.jobs_police)
-			dept_console = CRYO_SEC
-		else if(J.title in GLOB.jobs_medical)
+	if(job in SSjob.active_joinable_occupations)
+		job.free_job_positions(1)
+		if(ispolicejob(job))
+			dept_console = CRYO_SEC //TO DO: Turn this into a job var.
```

Comment:
Feel free to action later, but callout to the TODO:

Reply:
This will have to wait after this PR, due to the inheritance logic of the proc.
The department consoles are being determined not only by job but also by squad.